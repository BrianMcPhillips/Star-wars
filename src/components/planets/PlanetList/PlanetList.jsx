import React from 'react';
import PropTypes from 'prop-types';
import Planet from '../Planet/Planet';

const PlanetList = ({ data }) => {
  const planetStuff = data.map((planet, i) => 
    <li key={planet.url + i}>
      <Planet {...planet}/>
    </li>
  );

  return (
    <ul>
      {planetStuff}
    </ul>
  );

};

PlanetList.propTypes = {
  data: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    terrain: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
  })).isRequired
};

export default PlanetList;
