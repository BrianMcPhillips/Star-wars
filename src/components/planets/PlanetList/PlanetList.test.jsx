import React from 'react';
import { render, cleanup } from '@testing-library/react';
import PlanetList from './PlanetList';


describe('PlanetList component', () => {
  afterEach(() => cleanup());
  it('renders PlanetList', () => {
    const { asFragment } = render(
      <PlanetList data={[
        {
          name: 'Tatooine',
          terrain: 'desert',
          url: 'https://swapi.dev/api/planet/1/'
        },
        {
          name: 'Tatooine dos',
          terrain: 'desert cool',
          url: 'https://swapi.dev/api/planet/151515/'
        }
      ]}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
