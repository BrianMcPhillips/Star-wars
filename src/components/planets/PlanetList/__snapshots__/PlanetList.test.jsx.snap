// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanetList component renders PlanetList 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <h1>
        Tatooine
      </h1>
      <h4>
        desert
      </h4>
    </li>
    <li>
      <h1>
        Tatooine dos
      </h1>
      <h4>
        desert cool
      </h4>
    </li>
  </ul>
</DocumentFragment>
`;
